src/libopensc/pkcs15-pubkey.c

```diff
@@ -646,11 +646,17 @@ sc_pkcs15_decode_pubkey_ec(sc_context_t *ctx,
 	sc_copy_asn1_entry(c_asn1_ec_pointQ, asn1_ec_pointQ);
 	sc_format_asn1_entry(asn1_ec_pointQ + 0, &ecpoint_data, &ecpoint_len, 1);
 	r = sc_asn1_decode(ctx, asn1_ec_pointQ, buf, buflen, NULL, NULL);
-	if (r < 0)
+	if (r < 0) {
+		if (ecpoint_data)
+			free(ecpoint_data);
```

Comment:
the `if` check is not needed here as `free(NULL)` is completely valid. But please initialize the pointer to NULL on line 641 to make sure it is really initialized in all cases.

The same below.